src/lib.rs

```diff
@@ -743,6 +674,146 @@ fn send_tile_rebuild_tasks_system<C: OxidizedCollider>(
+        TypedShape::Cuboid(cuboid) => {
+            GeometryToConvert::Collider(ColliderType::Cuboid(*cuboid)).into()
+        }
+        TypedShape::Capsule(capsule) => {
+            GeometryToConvert::Collider(ColliderType::Capsule(*capsule)).into()
+        }
+        TypedShape::TriMesh(trimesh) => GeometryToConvert::ParryTriMesh(
+            // Can't turn these into boxed slices instantly because they are references.. So we need a copy.
+            trimesh.vertices().to_vec().into_boxed_slice(),
+            trimesh.indices().to_vec().into_boxed_slice(),
+        )
+        .into(),
+        TypedShape::HeightField(heightfield) => GeometryResult::Heightfield(heightfield),
+        TypedShape::ConvexPolyhedron(polyhedron) => {
+            let tri = polyhedron.to_trimesh();
+
+            GeometryToConvert::ParryTriMesh(tri.0.into_boxed_slice(), tri.1.into_boxed_slice())
+                .into()
+        }
+        TypedShape::Cylinder(cylinder) => {
+            GeometryToConvert::Collider(ColliderType::Cylinder(*cylinder)).into()
+        }
+        TypedShape::Cone(cone) => GeometryToConvert::Collider(ColliderType::Cone(*cone)).into(),
+        TypedShape::RoundCuboid(round_cuboid) => {
+            GeometryToConvert::Collider(ColliderType::Cuboid(round_cuboid.inner_shape)).into()
+        }
+        TypedShape::RoundCylinder(round_cylinder) => {
+            GeometryToConvert::Collider(ColliderType::Cylinder(round_cylinder.inner_shape)).into()
+        }
+        TypedShape::RoundCone(round_cone) => {
+            GeometryToConvert::Collider(ColliderType::Cone(round_cone.inner_shape)).into()
+        }
+        TypedShape::RoundConvexPolyhedron(round_polyhedron) => {
+            let tri = round_polyhedron.inner_shape.to_trimesh();
+
+            GeometryToConvert::ParryTriMesh(tri.0.into_boxed_slice(), tri.1.into_boxed_slice())
+                .into()
+        }
+        TypedShape::Triangle(triangle) => {
+            GeometryToConvert::Collider(ColliderType::Triangle(*triangle)).into()
+        }
+        TypedShape::RoundTriangle(triangle) => {
+            GeometryToConvert::Collider(ColliderType::Triangle(triangle.inner_shape)).into()
+        }
+        TypedShape::Compound(colliders) => {
+            let results = colliders
+                .shapes()
+                .iter()
+                .map(|(isometry, shape)| (*isometry, get_geometry_type(shape.0.as_typed_shape())))
+                .collect();
+
+            GeometryResult::Compound(results)
+        }
+        // These ones do not make sense in this.
+        TypedShape::HalfSpace(_) => GeometryResult::Unsupported, /* This is like an infinite plane? We don't care. */
+        TypedShape::Polyline(_) => GeometryResult::Unsupported,  /* This is a line. */
+        TypedShape::Segment(_) => GeometryResult::Unsupported,   /* This is a line segment. */
+        TypedShape::Custom(_) => {
+            warn!(
```

Comment:
This used to be `unimplemented!`, I believe `warn!` is a bit nicer in this case.